components/atom/input/src/Input/Features/Addon/index.scss

```diff
@@ -27,4 +34,33 @@ $c-atom-input-addon: $c-gray-light !default;
   &--left {
     border-width: $bdw-atom-input-addon-left;
   }
+
+  &--withIcon {
+    position: relative;
+    width: 100%;
+  
+    .sui-AtomInput-input {
```

Comment:
couldn't this be done via props in order to maintain it encapsulated?

Reply:
what do you suggest for this?